ssaHAL/ssa/core.py

```
@@ -42,6 +43,29 @@ def __fw_update_callback(_ssa: SSA, update_str: str):
+def __property_update_action(ssa: SSA, msg: string, property: string):
+    if property not in ssa.__properties:
+        print(f"[ERROR] Property `{property}` does not exist. Create it using `create_property` before setting it.")
+        return
+
+    value = json.loads(msg)
+    ssa.set_property(property, value)
+
+class ActDictElement():
+    # NOTE: The correct type here should have been `Callable[[SSA, str, Unpack[str]], None]`
+    # but the Unpack type is not yet supported in the typing module for MicroPython
+    self.callback: Optional[Callable[[SSA, str, Any], None]]
```

Ruff (0.8.2)

Undefined name `self`

(F821)